tensorboard/webapp/settings/_views/dialog_component.ts

```diff
@@ -177,7 +159,29 @@ export class SettingsDialogComponent implements OnInit, OnDestroy {
     this.ngUnsubscribe.complete();
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['reloadPeriodInMs']) {
+      const change = changes['reloadPeriodInMs'];
+      if (change.previousValue !== change.currentValue) {
+        this.reloadPeriodControl.setValue(change.currentValue / 1000);
+      }
+    }
+
+    if (changes['reloadEnabled']) {
+      const change = changes['reloadEnabled'];
+      if (change.currentValue) this.reloadPeriodControl.enable();
+      else this.reloadPeriodControl.disable();
+    }
+
+    if (changes['pageSize']) {
+      const change = changes['pageSize'];
+      if (change.previousValue !== change.currentValue) {
+        this.paginationControl.setValue(change.currentValue);
+      }
+    }
```

Comment:
is it possible to have multiple changes at a time?
i.e., line 163 and 170 are both true

if not, can we say
```
const change  = changes['reloadPeriodInMs'] ?? (changes['reloadEnabled'] ?? (changes['pageSize']) ?? null));
if (!change) return;
```

Reply:
In theory it could happen that you have multiple changes at the same time.